packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/Converters.kt

```diff
@@ -179,21 +262,55 @@ internal val primitiveTypeConverters: Map<KClass<*>, RealmValueConverter<*>> =
         Short::class to ShortConverter,
         Int::class to IntConverter,
         RealmInstant::class to RealmInstantConverter,
+        RealmInstantImpl::class to RealmInstantConverter,
+        BsonObjectId::class to ObjectIdConverter,
         ObjectId::class to RealmObjectIdConverter,
         ObjectIdImpl::class to RealmObjectIdConverter,
         RealmUUID::class to RealmUUIDConverter,
-        ByteArray::class to ByteArrayConverter
-    ).withDefault { StaticPassThroughConverter }
+        RealmUUIDImpl::class to RealmUUIDConverter,
+        ByteArray::class to ByteArrayConverter,
+        String::class to StaticPassThroughConverter,
+        Long::class to StaticPassThroughConverter,
+        Boolean::class to StaticPassThroughConverter,
+        Float::class to StaticPassThroughConverter,
+        Double::class to StaticPassThroughConverter,
+    )
 
 // Dynamic default primitive value converter to translate primary keys and query arguments to RealmValues
+@Suppress("NestedBlockDepth")
 internal object RealmValueArgumentConverter {
-    fun convertArg(value: Any?): RealmValue {
+    fun MemAllocator.convertArg(value: Any?): RealmValue {
```

Comment:
There is something wrong in the typesystem because we cannot import any argument without tracking memory.